lib/metrician/reporters/memcache.rb

```diff
@@ -16,7 +16,7 @@ def client_classes
       [
         self.class.memcached_gem? && Memcached,
         self.class.dalli_gem? && Dalli::Client,
-      ].compact
+      ].select{|x| x}
```

Comment:
At this point I think this would probably be better re-written as an `if/elsif/else`? What do you think? We could also go back on making the helper methods `#memcached_gem?` and `#dalli_gem?` return actual boolean values, since `defined?(ThingThatIsNotDefined)` returns `nil`.